scylla/tests/integration/tablets.rs

```diff
@@ -0,0 +1,628 @@
+        Err(err) => panic!("{}", err),
+    }
+}
+
+#[tokio::test]
+#[ntest::timeout(30000)]
+async fn test_tablet_feedback_received_with_default_policy() {
+    setup_tracing();
+    const TABLET_COUNT: usize = 16;
+
+    let res = test_with_3_node_cluster(
+        ShardAwareness::QueryNode,
+        |proxy_uris, translation_map, mut running_proxy| async move {
+            let session = scylla::SessionBuilder::new()
+                .known_node(proxy_uris[0].as_str())
+                .address_translator(Arc::new(translation_map))
+                .build()
+                .await
+                .unwrap();
+            let ks = unique_keyspace_name();
+
+            /* Prepare schema */
+            prepare_schema(&session, &ks, "t", TABLET_COUNT).await;
+
+            let tablets = get_tablets(&session, ks.clone(), "t".to_string()).await;
+
+            let prepared = session
+                .prepare(format!(
+                    "INSERT INTO {}.t (a, b, c) VALUES (?, ?, 'abc')",
+                    ks
+                ))
+                .await
+                .unwrap();
+
+            let value_lists = calculate_key_per_tablet(&tablets, &prepared);
+
+            let (feedback_txs, mut feedback_rxs): (Vec<_>, Vec<_>) = (0..3)
+                .map(|_| mpsc::unbounded_channel::<(ResponseFrame, Option<TargetShard>)>())
+                .unzip();
+            for (i, tx) in feedback_txs.iter().cloned().enumerate() {
+                running_proxy.running_nodes[i].change_response_rules(Some(vec![ResponseRule(
+                    Condition::ResponseOpcode(ResponseOpcode::Result)
+                        .and(Condition::not(Condition::ConnectionRegisteredAnyEvent)),
+                    ResponseReaction::noop().with_feedback_when_performed(tx),
+                )]));
+            }
+
+            // We can't expect feedback for each query, because replica list may
+            // be the same for given token in tablets and in token ring, but we
+            // can expect to receive some feedbacks.
+            let mut statements_with_feedback = 0;
+            for values in value_lists.iter() {
+                info!(
+                    "First loop, trying key {:?}, token: {}",
+                    values,
+                    prepared.calculate_token(&values).unwrap().unwrap().value()
+                );
+                try_join_all((0..100).map(|_| async { session.execute(&prepared, values).await }))
+                    .await
+                    .unwrap();
+                let feedbacks: usize = feedback_rxs.iter_mut().map(count_tablet_feedbacks).sum();
+                if feedbacks > 0 {
+                    statements_with_feedback += 1;
+                }
+            }
+
+            // There is no special meaning to "3", I just thought it gives low
+            // enough chances of failure.
+            assert!(statements_with_feedback > (value_lists.len() / 3));
```

Comment:
No specific reason and no calculations, just intuition

Reply:
Hm... I read my comment above now and I have no idea where I got "30" from.
But other change (when there is no tablet in local data then return empty replica list, not vnode replica list) should make this test totally non-flaky, even when expecting to receive feedback for all tablets, not a subset of them.

Reply:
I've run the test a 100 times after changes (requiring that always feedbacks > 0), so I can merge it with `test_tablet_awareness_works_with_full_info` because the workaround with one-shot LBPs is no longer necessary,